### PR TITLE
add edge-case-handling for `subsets`

### LINKED ISSUE #1039: Edge-cases for `subsets`
All implementations for subsets of a given size, namely 
* [`subsets(::Int, k::Int)`](https://github.com/thofma/Hecke.jl/blob/46b30aaae5ae96a43196f8de45e01f8cc4edacbe/src/NumFieldOrd/NfOrd/LLL.jl#L433)
* [`subsets(::Vector{T}, k::Int)`](https://github.com/thofma/Hecke.jl/blob/46b30aaae5ae96a43196f8de45e01f8cc4edacbe/src/NumFieldOrd/NfOrd/LLL.jl#L457)
* [`subsets_it(::Int, k::Int)`](https://github.com/thofma/Hecke.jl/blob/46b30aaae5ae96a43196f8de45e01f8cc4edacbe/src/NumFieldOrd/NfOrd/LLL.jl#L412)
* [`subsets(::Set{T}, k::Int)`](https://github.com/thofma/Hecke.jl/blob/46b30aaae5ae96a43196f8de45e01f8cc4edacbe/src/Misc/MSet.jl#L273)
 
fail for most cases, where `k` is larger than the first argument. The exception seems to be `subsets(0,1)` and `Hecke.subsets_it(0,1)`.
 
I would expect all of them to return an empty list instead of failing. For its use as a replacement for `Combinatorics.combinations` in `Oscar.jl, this would be beneficial as well.

I further noticed, that there are no tests for all of these, so I would suggest adding those including tests for the above problems.

### PR DESCRIPTION
**Pull request by lgoettgens**
Resolves #1039.

### CODE REVIEW

**Comment by lgoettgens** [comment]
If requested, I can add some tests for the different `subsets` methods. I am however unsure, of where to put them. The file structure of Hecke looks rather intimidating to me.

File: src/Misc/MSet.jl [hunk 1]
```diff
@@ -94,29 +94,33 @@ end
...
+  return u
 end
-Base.union!(s::MSet, xs) = (for x=xs; push!(s,x); end; s)
-Base.union!(s::MSet, xs::AbstractArray) = (for x=xs; push!(s,x); end; s)
+Base.union!(s::MSet, xs) = (for x = xs
+  push!(s, x)
+end; s)
+Base.union!(s::MSet, xs::AbstractArray) = (for x = xs
+  push!(s, x)
+end; s)
```
[the remark this answers is not in the document]
Comment (lgoettgens) [reply src/Misc/MSet.jl:109]: oh, I must have re-formatted the file by mistake. Sorry, will force-push in a minute.

**Comment by thofma** [comment]
I think it is fine to skip the tests.

**Comment by lgoettgens** [comment]
I think the test failures are independent of my changes, right?

**Pull request merged**